crates/optimism/txpool/src/validator.rs

```diff
@@ -207,11 +242,61 @@ where
     /// Returns all outcomes for the given transactions in the same order.
     ///
     /// See also [`Self::validate_one`]
-    pub fn validate_all(
+    pub async fn validate_all(
         &self,
         transactions: Vec<(TransactionOrigin, Tx)>,
     ) -> Vec<TransactionValidationOutcome<Tx>> {
-        transactions.into_iter().map(|(origin, tx)| self.validate_one(origin, tx)).collect()
+        futures_util::future::join_all(
+            transactions.into_iter().map(|(origin, tx)| self.validate_one(origin, tx)),
+        )
+        .await
+    }
+
+    /// Extracts commitment from access list entries, pointing to 0x420..022 and validates them
+    /// against supervisor.
+    ///
+    /// If commitment present pre-interop tx rejected.
+    ///
+    /// Returns:
+    /// None - if tx is not cross chain,
+    /// Some(Ok(true)) - if tx is valid cross chain,
+    /// Some(Ok(false)) - if tx is cross chain but interop is not active,
+    /// Some(Err(e)) - if tx is not valid
+    pub async fn is_valid_cross_tx(&self, tx: &Tx) -> Option<Result<bool, InvalidCrossTx>> {
+        // We don't need to check for deposit transaction in here, because they won't come from
+        // txpool
+        let access_list = tx.access_list()?;
+        let inbox_entries =
+            SupervisorClient::parse_access_list(access_list).copied().collect::<Vec<_>>();
+        if inbox_entries.is_empty() {
+            return None;
+        }
+        let timestamp = self.block_info.timestamp.load(Ordering::Relaxed);
+        // Interop check
+        if self.fork_tracker.is_interop_activated() {
+            // No cross chain tx allowed before interop
+            return Some(Ok(false))
+        }
+        let client = self
+            .supervisor_client
+            .as_ref()
+            .expect("supervisor client should be always set after interop is active");
+        let safety_level = client.safety();
+        match client
+            .validate_messages(
+                inbox_entries.as_slice(),
+                safety_level,
+                ExecutingDescriptor::new(timestamp, Some(86400)),
```

Comment:
what is this constant here?

Reply:
1 day, this is just for now, i will make it configurable, once we will have field in pool transaction

Reply:
moved in into the constant with the docstring